### ISSUE TITLE
Add traits for Did Linked Resources

### ISSUE THREAD

**Issue by sownak** [decentralized-identity/did-traits#21]
With `createResource()`, `updateResource()` and `deactivateResource()` support now enabled, we should support the did resource traits as well. 
Example definitions:
```json
{
    "updateableResource": {
      "type": "boolean",
      "title": "Rotatable Did linked resources",
      "description": "Did linked resources are updateable, allowing the resources to be replaced or updated with newer versions."
    },
    "deactivatableResource": {
      "type": "boolean",
      "title": "Deactivatable Did linked resources",
      "description": "Did linked resources are deactivatable."
    },
}
```

**Comment by jceb** [comment]
@sownak thank you for the suggestion. I'm not familiar with DID Linked Resources. Could you provide a brief overview of what needs to be done for a DID method to support linked resources? In the introduction (https://w3c-ccg.github.io/DID-Linked-Resources/#intro) it sounds like this a feature of the DID resolver rather than a characteristic of a DID method.

**Comment by sownak** [comment]
@jceb Yes the DID-Linked resources are (as explained in the link) files/objects which are identified and retrieved by a DID. For a did method to support linked resource, there would be additional metadata to be added like `linkedResourceMetadata`, the CRUD operations for the resources need to be handled, etc. DID methods like did:cheqd support this, and in https://identity.foundation/did-registration/#did-url-operations the methods that I mentioned are defined as well.

**Comment by jceb** [comment]
If I understand correctly, the trait refers to additional DID URL operations as defined in https://identity.foundation/did-registration/#did-url-operations. Is it important to name every CRUD operation as a trait or are DID methods only required to implement a subset of the operation, e.g. only create and deactivate?

Is DID Linked Resources a concrete specification DID URL operations on resources, basically an extension to the DID Registration specification?

**Comment by jceb** [comment]
For reference: DID Linked Resources spec: https://w3c-ccg.github.io/DID-Linked-Resources/
